core/templates/pages/topics-and-skills-dashboard-page/skills-list/skills-list.component.spec.ts

```diff
@@ -68,9 +68,14 @@ describe('Skills List Component', () => {
 
 
   class MockNgbModal {
-    modal: string;
+    modal!: string;
```

Comment:
Why is the `!` needed here?

Reply:
it is used for non null assertion here . Typescript wants var `modal` to have an initialization . But here in this class we have done the initialization indirectly in the if else decision structure . Typescript is not able to catch those indirect initialization and thus sends a ( no initializer ) error . To overcome this , we can use this `!` operator to convey that we have done the initialization afterwards in the current scope , so that now typescript will allow this .

Reply:
Please add an explanatory code comment for this.

Reply:
ohh sure !!

Reply:
@vojtechjelinek will this line be adequate ?
`modal!: string; // Avoid non null assertion for modal by suffixing ! symbol;`

Reply:
You should explain why we need to avoid the non-null assertion.

Reply:
ptal @vojtechjelinek